docs/architecture/commitlog/fileformat.rst

```diff
@@ -0,0 +1,57 @@
+Scylla commitlog segment file format
+====================================
+
+File descriptor structure
+.........................
+
+Scylla commitlog segment files are named with a versioned, time-indexed scheme, as::
+
+        <Prefix><version>-<id>.log
+
+Where `<Prefix>` is application specific, but typically "Commitlog-", or in case of 
+files being recycled "Recycled-Commitlog-", `<version>` is the file format version,
+and `<id>` is the id part of a replay position (timestamp + shard).
+
+Segment file data structure
+...........................
+
+All control data is written in network byte order.
+
+The file consists of a file header, followed by any number of chunks. Each chunk has 
+its own header + a marker to the start of next chunk, to allow skipping it more easily,
+should any data corruption be present in the chunks data.
+
+Chunks contain data entries, with a small header, stored data + checksums to verify its integrity.
+
+An entry can be a "multi-entry", i.e. several entries written as one.
+
+Version 2
```

Comment:
Not clear what this is? Is this the current version we use?

Reply:
Added blurb. Yes.